transport/api/src/lib.rs

```diff
@@ -517,6 +519,17 @@ where
         )
         .await;
 
+        if let Some(port) = self.cfg.protocol.autonat_port {
+            transport_layer.run_nat_server(port);
+        }
+
+        if addresses.is_empty() {
+            warn!("No addresses found in the database, not dialing any NAT servers");
+        } else {
+            info!(num_addresses = addresses.len(), "Found addresses from the database");
```

Comment:
I would probably check for duplicates here (`sort` + `dedup`)
and then randomized the array again (`shuffle`)